components/track/TrackList.vue

```diff
@@ -13,9 +13,22 @@ withDefaults(
     skeletonCount: 5,
   }
 );
+
 const { setCurrentTrack, addTrackToQueue } = inject(MediaPlaylistInjectionKey)!;
 const showDropDownForTrack: Ref<null | string> = ref(null);
 
+const isTrackTypeKnown = () => {
+  const firstType = props.tracks?.[0]?.subtype;
+  console.log("first type", firstType);
+  const tracksWithDifferentType = props.tracks?.filter(
```

Comment:
Just a minor note: If you're not interested in the actual amount, why not use [`find()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) or even better: [`every()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) or [`some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)? Any of them would stop after the first occurrence - the latter make the code even more intuitive to read IMO.

Maybe selecting a matcher-function based on the first type makes it even better to read - at least quicker to run, but I'll leave this over to you 😄

Reply:
I didn't remember what functions are available and what they are called. Thanks for the hint.